Lib/test/test_dataclasses.py

```diff
@@ -2767,6 +2767,21 @@ class C:
             c.i = 5
         self.assertEqual(c.i, 10)
 
+    def test_frozen_empty(self):
+        @dataclass(frozen=True)
+        class C:
+            pass
+
+        c = C()
+        self.assertFalse(hasattr(c, 'i'))
+        with self.assertRaises(FrozenInstanceError):
+            c.i = 5
+        self.assertFalse(hasattr(c, 'i'))
+        with self.assertRaises(FrozenInstanceError):
+            del c.i
+        with self.assertRaises(AttributeError):
+            del c.i
```

Comment:
I think we can leave out these two lines. If we need a test that `FrozenInstanceError` is a subclass of `AttributeError`, `test_frozen_empty` is an odd place to put that test. It doesn't seem related to the main purpose of the test.